### PR TITLE
 [13.0] [MIG] purchase_order_line_description

### PR DESCRIPTION
**Pull request by Nikul-Chaudhary**
Purchase order line description
===============================
This module allows to use only the product purchase description on the purchase
order lines.

### CODE REVIEW

File: purchase_order_line_description/models/res_config_settings.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+from odoo import fields, models
+
+
+class ResConfigSettings(models.TransientModel):
+    _name = 'res.config.settings'
+    _inherit = 'res.config.settings'
+
+    group_use_product_description_per_po_line = fields.Selection(
+        selection=[
+            (0, "Use the product name on the purchase order lines"),
```
Comment (sudhir-erpharbor) [inline purchase_order_line_description/models/res_config_settings.py]: Please convert the selection keys to string as v13 does not allow int keys in the selection.

File: purchase_order_line_description/README.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+   out that now purchase order line description only contains the description
+   you are created.
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/purchase-workflow/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed
+`feedback <https://github.com/OCA/purchase-workflow/issues/new?body=module:%20purchase_order_line_description%0Aversion:%2012.0%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (sudhir-erpharbor) [inline purchase_order_line_description/README.rst]: 13.0

**Comment by Nikul-Chaudhary** [comment]
@sudhir-erpharbor @aheficent @cubells  can you please review it?

**Comment by AaronHForgeFlow** [comment]
runbot is red :cry:

**Comment by sudhir-erpharbor** [comment]
Code review LGTM. Not sure why runbot is red.

File: purchase_order_line_description/models/res_config_settings.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+
+class ResConfigSettings(models.TransientModel):
+    _name = "res.config.settings"
+    _inherit = "res.config.settings"
+
+    group_use_product_description_per_po_line = fields.Selection(
+        selection=[
+            ("0", "Use the product name on the purchase order lines"),
+            ("1", "Use the product purchase description on the purchase" "order lines"),
```
Comment (kittiu) [inline purchase_order_line_description/models/res_config_settings.py:15]: Just the comment to "Black". I am not sure why it create space between purchase and order. :p

File: purchase_order_line_description/models/purchase_order_line.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.onchange("product_id")
+    def onchange_product_id(self):
```
Comment (kittiu) [inline purchase_order_line_description/models/purchase_order_line.py:12]: As stated in coding guideline
```suggestion
    def _onchange_product_id(self):
```

File: purchase_order_line_description/tests/test_purchase_order_line_description.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        super(TestPurchaseOrderLineDescription, cls).setUpClass()
+        partner = cls.env["res.partner"].create({"name": "Test partner"})
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Product",
+                "standard_price": 10,
+                "description_purchase": "description for purchase",
+            }
+        )
+        group_id = cls.env.ref(
```
Comment (kittiu) [inline purchase_order_line_description/tests/test_purchase_order_line_description.py]: As in coding guideline
https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#342variable-names
```suggestion
        group = cls.env.ref(
```

File: purchase_order_line_description/tests/test_purchase_order_line_description.py [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+        cls.test_user = cls.env["res.users"].create(
+            {
+                "name": "test",
+                "login": "test",
+                "groups_id": [(6, 0, [res_users_purchase_user.id])],
+            }
+        )
+        # add group_use_product_description_per_po_line to test_user
+        cls.test_user.write({"groups_id": [(4, group_id.id)]})
+        product_uom_id = cls.env.ref("uom.product_uom_unit")
```
Comment (kittiu) [inline purchase_order_line_description/tests/test_purchase_order_line_description.py]:
```suggestion
        product_uom = cls.env.ref("uom.product_uom_unit")
```

**Review by kittiu — APPROVED** [review]
With some comments.
Otherwise, LGTM

**Comment by Nikul-Chaudhary** [comment]
@AaronHForgeFlow @sudhir-erpharbor can you please update your review?

File: purchase_order_line_description/models/res_config_settings.py (hunk 3 above)
Comment (HviorForgeFlow) [inline purchase_order_line_description/models/res_config_settings.py:15]:
```suggestion
            ("1", "Use the product purchase description on the purchase order lines"),
```

File: purchase_order_line_description/models/purchase_order_line.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        res = super(PurchaseOrderLine, self).onchange_product_id()
```
Comment (HviorForgeFlow) [inline purchase_order_line_description/models/purchase_order_line.py:13]:
```suggestion
        res = super().onchange_product_id()
```

File: purchase_order_line_description/models/purchase_order_line.py (hunk 4 above)
Comment (HviorForgeFlow) [reply purchase_order_line_description/models/purchase_order_line.py:12]: It can be kept as it is as has the same name as used in their inheritance.

**Review by HviorForgeFlow — CHANGES_REQUESTED** [review]

**Comment by rousseldenis** [comment]
@Nikul-Chaudhary

**Pull request closed without merging**